wire-ios-request-strategy/Sources/Request Strategies/Assets/Helpers/AssetDownloadRequestFactory.swift

```diff
@@ -19,25 +19,25 @@
 import Foundation
 import WireTransport
 
-public final class AssetDownloadRequestFactory: NSObject {
+public final class AssetDownloadRequestFactory {
 
     public func requestToGetAsset(withKey key: String, token: String?, domain: String?, apiVersion: APIVersion) -> ZMTransportRequest? {
-        let path: String
 
+        let domain = if let domain, !domain.isEmpty { domain } else { BackendInfo.domain }
+        let path: String
         switch apiVersion {
         case .v0:
             path = "/assets/v3/\(key)"
         case .v1:
-            guard let domain = domain.nonEmptyValue ?? BackendInfo.domain else { return nil }
+            guard let domain else { return nil }
```

Comment:
I think it's not 100% the same. We also check `BackendInfo.domain` if domain is nil.